src/octoprint/plugins/virtual_printer/__init__.py

```diff
@@ -178,9 +185,10 @@ def _copy_to_incoming(self):
 
                 try:
                     while True:
-                        try:
+                        with self._mutex:
```

Comment:
This forces half-duplex on *every* wrapper. I'm not sure that's the correct solution here.

Reply:
You are most certainly right. No need for this. Removed